cpp/src/aggregation/aggregation.cpp

```diff
@@ -155,10 +155,13 @@ std::unique_ptr<aggregation> make_udf_aggregation(udf_type type,
 namespace detail {
 namespace {
 struct target_type_functor {
+  data_type type;
   template <typename Source, aggregation::Kind k>
   constexpr data_type operator()() const noexcept
   {
-    return data_type{type_to_id<target_type_t<Source, k>>()};
+    auto const id = type_to_id<target_type_t<Source, k>>();
+    return id == type_id::DECIMAL32 || id == type_id::DECIMAL64 ? data_type{id, type.scale()}
+                                                                : data_type{id};
```

Comment:
why skip scale for non-decimal type? (It will have default value, anyway it won't be used).

Reply:
Ctor with scale only designed to be called for Decimal types.
```
  explicit data_type(type_id id, int32_t scale) : _id{id}, _fixed_point_scale{scale}
  {
    assert(id == type_id::DECIMAL32 || id == type_id::DECIMAL64);
  }
```